CMakeLists.txt

```diff
@@ -169,10 +169,13 @@ include( GNUInstallDirs )
 include( TestBigEndian )
 
 cmake_dependent_option(
-   ${_OPT}has_sentry_reporting 
+   ${_OPT}has_sentry_reporting
    "Build support for sending errors to Sentry"
    On
-   "${_OPT}has_networking;DEFINED SENTRY_DSN_KEY;DEFINED SENTRY_HOST;DEFINED SENTRY_PROJECT"
+   "${_OPT}has_networking;\
+      NOT \"$ENV{SENTRY_DSN_KEY}\" STREQUAL \"\";\
```

Comment:
I'm not sure if I like this approach. How about something like

```
if( NOT DEFINED SENTRY_DSN_KEY AND NOT "$ENV{SENTRY_DSN_KEY}" STREQUAL "" )
    set( SENTRY_DSN_KEY  $ENV{SENTRY_DSN_KEY} )
endif()
```

This may sound like overkill, but I already had to help Paul and David Bailes setting up the local build. Giving a specific line is much easier for me and for them to do, rather than messing with the system environment.  Also, I prefer not to pollute the environment on developers' machines.

Reply:
Also, we can easily append -D options to cmake in configure.sh.

Github will remove it from the build log anyway.